src/App/App.js

```diff
@@ -28,23 +28,52 @@ function App() {
       console.log(error);
     }
   }
+  
+  function handleDelete(selectedSpell){
+    function deleteSpells() {
+      console.log('spell deleted');
+      selectedSpell.isKnown = false;
+      const filteredSpells = savedSpells.filter(spell => spell.id !== selectedSpell.id);
+      console.log(filteredSpells, 'filtered')
+      setSavedSpells(filteredSpells);
+      console.log(savedSpells, "deleted")
+      return savedSpells;
+    };
+    savedSpells.length > 0 && deleteSpells();
+    return savedSpells;
+  }
+
+  function handleKnown(selectedSpell) {
+    function saveSpells() {
+      console.log('spell saved');
+      selectedSpell.isKnown = true;
+      savedSpells.push(selectedSpell)
+      console.log(savedSpells, "saved")
+      return selectedSpell;
+    }; 
+      
+    !selectedSpell.isKnown && saveSpells();
+    console.log(selectedSpell, 'selectedSpell')
+    return savedSpells;
+  }
 
   useEffect(() => {
     getSpells()
-    .then(setSpells);
+    .then(setSpells)
   }, []);
 
   return (
     <div className="App">
       <header className="App-header">
         <h1>spellkeeper</h1>
       </header>
-      <Main spells={spells}/>
+      {spells.length === 0 && <span className='message'>loading spells</span>}
```

Comment:
I wont lie, i saw this was the way you put in your loading message and I did the exact same logic for my loading as well so thank you for this